addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+            else:
+                raise RuntimeError("Invalid matrix subtype")
+            # Validate that each line has been answered
+            if answer_number != lines_number:
+                errors.update({answer_tag: question.constr_error_msg})
+        return errors
+
+
+class SurveyLabel(models.Model):
+    ''' A suggested answer for a question '''
+    _name = 'survey.label'
+    _rec_name = 'value'
+    _order = 'sequence,id'
+    _description = 'Survey Label'
+
+    question_id = fields.Many2one('survey.question', string='Question',
+        ondelete='cascade')
+    question_id_2 = fields.Many2one('survey.question', string='Question',
+        ondelete='cascade')
+    sequence = fields.Integer(string='Label Sequence order', default=10)
+    value = fields.Char(string="Suggested value", translate=True,
+        required=True)
+    quizz_mark = fields.Float(string='Score for this choice', help="A positive score indicates a correct choice; a negative or null score indicates a wrong answer")
+
+    @api.multi
+    @api.constrains('question_id', 'question_id_2')
+    def _check_question_not_empty(self):
+        '''Ensure that field question_id XOR field question_id_2 is not null'''
+        for label in self:
+            # 'bool()' is required in order to make '!=' act as XOR with objects
+            if not bool(label.question_id) != bool(label.question_id_2):
+                raise UserError(_("A label must be attached to one and only one question"))
+
+class SurveyUserInput(models.Model):
+    ''' Metadata for a set of one user's answers to a particular survey '''
+    _name = "survey.user_input"
+    _rec_name = 'date_create'
+    _description = 'Survey User Input'
+
+    survey_id = fields.Many2one('survey.survey', string='Survey', required=True,
+                                 readonly=True, ondelete='restrict')
+    date_create = fields.Datetime(string='Creation Date', required=True,
+                                   readonly=True, copy=False, default=fields.Datetime.now)
+    deadline = fields.Datetime(string="Deadline",
+                            help="Date by which the person can open the survey and submit answers",
+                            oldname="date_deadline")
+    input_type = fields.Selection([('manually', 'Manually'), ('link', 'Link')],
+                             string='Answer Type', required=True, readonly=True,
+                             oldname="response_type", default='manually')
+    state = fields.Selection([('new', 'Not started yet'),
+                               ('skip', 'Partially completed'),
+                               ('done', 'Completed')],
+                              string='Status',
+                              readonly=True, default='new')
+    test_entry = fields.Boolean(readonly=True)
+    token = fields.Char(string="Identification token", readonly=True, required=True, copy=False, default=lambda self: uuid.uuid4().__str__())
```

Comment:
Agree, Its not function of current object, so it will not work(I didn't check default_get :))